### PR TITLE
Move eligibility gating toggle to be with the rest of the program settings

### LINKED ISSUE #5852: Gating eligibility flag is hard to find
**Describe the bug**
In order to turn on / off gating eligibility, you have to click on the "program settings" link in the eligibility section of the page: 

Screenshot 2023-10-24 at 4 30 03 PM

This should appear in the program details section, like other settings.

### PR DESCRIPTION
**Pull request by caitlinshk**
### Description

This moves the eligibility gating toggle to be with the rest of the program settings.

Previously, it was on a separate settings menu that was only accessible from these places:

**Program dashboard -> overflow menu on draft program -> "Settings":**
  
image

**Program edit blocks page -> "program settings" link inside the helper text for eligibility conditions:**

image
  

Neither of those locations are obvious to admins. This PR moves it to be with the rest of the program-level settings so it's easier to access.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

### Instructions for manual testing

1. Create a new program, set its eligibility to **non-gating**. Create a question with an eligibility condition. As an applicant, fill out the program with an eligible answer and verify you can submit the form. Fill out the form with a *not* eligible answer and verify you can still submit the form.
2. Create a new program, set its eligibility to **gating**. Create a question with an eligibility condition. As an applicant, fill out the program with an eligible answer and verify you can submit the form. Fill out the form with a *not* eligible answer and verify you get the "your answers aren't eligible" page and you cannot submit the form.
3. Change an existing program from gating to non-gating or vice versa. Verify the form behavior changes appropriately.

### Issue(s) this completes

Fixes #5852

### CODE REVIEW

**Comment by caitlinshk** [comment]
* **#6950** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6950?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @caitlinshk and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: browser-test/src/applicant/application_navigation.test.ts [hunk 1]
```diff
@@ -1515,8 +1521,11 @@ test.describe('Applicant navigation flow', () => {
 
         await expect(addressCorrectionInput).toHaveValue('true')
 
-        // Set thing to soft eligibilty
-        await adminPrograms.toggleEligibilityGating()
+        // Set program to soft eligibility
```
Comment (rockycodes) [inline browser-test/src/applicant/application_navigation.test.ts:1524]: nitpick: I don't think you need this comment anymore since it's clear it's being set to to not gating

File: server/app/views/admin/programs/ProgramFormBuilder.java [hunk 2]
```diff
@@ -181,7 +185,36 @@ private FormTag buildProgramForm(
...
+        showTiSelectionList(selectedTi, displayMode.equals(DisplayMode.SELECT_TI.getValue())));
+
+    formTag.with(
+        // TODO(#2618): Consider using helpers for grouping related radio controls.
+        fieldset()
+            .with(
+                legend("Program eligibility gating")
+                    .withClass(BaseStyles.INPUT_LABEL)
+                    .with(ViewUtils.requiredQuestionIndicator())
+                    .with(p("(Not applicable if this program is the pre-screener)")),
```
Comment (rockycodes) [inline server/app/views/admin/programs/ProgramFormBuilder.java:197]: suggestion (blocking): This text might be confusing since some CEs might not be using pre-screeners. I suggest conditionally showing this text if the prescreener feature flag is enabled and/or only showing this section when the pre-screener button is NOT checked.

File: server/app/views/admin/programs/ProgramFormBuilder.java [hunk 3]
```diff
@@ -181,7 +185,36 @@ private FormTag buildProgramForm(
...
+                FieldWithLabel.radio()
+                    .setFieldName(ELIGIBILITY_IS_GATING_FIELD_NAME)
+                    .setAriaRequired(true)
+                    .setLabelText(
+                        "Eligibility criteria blocks submission (applicants must meet all"
```
Comment (rockycodes) [inline server/app/views/admin/programs/ProgramFormBuilder.java:202]: suggestion (blocking): My brain had a hard time parsing the meaning of this sentence. That could just be because eligibility stuff is a brain twister and takes some extra thinking, but it would be great to get the sign off on this from design and/or product. (update: we chatted about this in Slack and Sijia will take a look when they're back)

**Review by rockycodes — CHANGES_REQUESTED** [review]
Overall this looks great! I really love this change -- the code is cleaner and more intuitive and the UI is too! 

I have a couple of comments around language/user interface.

File: browser-test/src/applicant/application_navigation.test.ts (hunk 1 above)
Comment (caitlinshk) [reply browser-test/src/applicant/application_navigation.test.ts]: Updated!

File: server/app/views/admin/programs/ProgramFormBuilder.java (hunk 2 above)
Comment (caitlinshk) [reply server/app/views/admin/programs/ProgramFormBuilder.java]: Oh that's a great callout! Updated to only show if the flag is enabled.

File: server/app/views/admin/programs/ProgramFormBuilder.java (hunk 3 above)
Comment (caitlinshk) [reply server/app/views/admin/programs/ProgramFormBuilder.java:204]: Hah, I tested this text on https://hemingwayapp.com/ and it got a reading level of "post-graduate" 👎 Re-writing to:
- Residents can only submit applications if they meet all the eligibility requirements
- Residents can submit applications even if they don't meet eligibility requirements

takes it down to grades 13/14, which still isn't great but better.

Removing "eligibility requirements" takes it down to grade 7, so I think that phrase will always spike the rating, but that's also the phrasing we use everywhere. I'll run the wording by Sijia!

**Comment by caitlinshk** [comment]
Note: I updated `browser-test/image_snapshots/admin_program_creation_test/program-creation-page.png` to use new wording but not the rest of them -- will only update once I get the okay from Sijia.

**Review by rockycodes — APPROVED** [review]
this looks good to me pending Sijia's approval of language!

File: server/app/models/ProgramModel.java [hunk 4]
```diff
@@ -211,7 +212,7 @@ public ProgramModel(
     this.statusDefinitions = new StatusDefinitions();
     this.versions.add(associatedVersion);
     this.programType = programType;
-    this.eligibilityIsGating = true;
+    this.eligibilityIsGating = eligibilityIsGating;
```
[the remark this answers is not in the document]
Comment (caitlinshk) [reply server/app/models/ProgramModel.java:215]: Yep, it was defaulted to always be true! It could changed to false if you used `ProgramDefinition.Builder.setEligibilityIsGating(false)` and then built a new `ProgramModel` from that `ProgramDefinition`.

[the remark this answers is not in the document]
Comment (azhangggggg) [reply server/app/models/ProgramModel.java:215]: Got it. Thanks!

**Review by azhangggggg — APPROVED** [review]

**Comment by sijiayam** [comment]
@caitlinshk The radio buttons looks great.

Suggest rewording, just trying to frame it with more actionable wording:
- Only allow residents to submit applications if they meet all eligibility requirements
- Allow residents to submit applications even if they don't meet eligibility requirements
